cert/openssl.cnf

```diff
@@ -135,260 +140,366 @@ caIssuers;URI.0 = http://cacerts.digicert.com/WinnForumCA.crt
 
-ROLE=ASN1:SEQUENCE:pi_seq
+[ cps ]
 
-[ pi_seq ]
-field1 = UTF8:INSTALLER
-field2 = UTF8:CA
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
 
-[ wif_pi_ca_sign ]
+[ pi_ca_sign ]
 
 subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid:always,issuer
 basicConstraints = critical, CA:true, pathlen:0
 keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_INSTALLER
 
-ROLE=ASN1:SEQUENCE:pi_seq
+[ cps ]
 
-[ pi_seq ]
-field1 = UTF8:INSTALLER
-field2 = UTF8:CA
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
 
 ####################################################################
-[ wif_pal_ca ]
+[ pal_ca ]
 
 subjectKeyIdentifier = hash
 basicConstraints = critical, CA:true, pathlen:0
 keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_PAL
 
-ROLE=ASN1:SEQUENCE:pal_seq
+[ cps ]
 
-[ pal_seq ]
-field1 = UTF8:PAL
-field2 = UTF8:CA
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
 
-[ wif_pal_ca_sign ]
+[ pal_ca_sign ]
 
 subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid:always,issuer
 basicConstraints = critical, CA:true, pathlen:0
 keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_PAL
+
+[ cps ]
+
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
+
+####################################################################
+[ cbsd_ca ]
+
+subjectKeyIdentifier = hash
+basicConstraints = critical, CA:true, pathlen:1
+keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_CBSD
+
+[ cps ]
+
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
+
+[ cbsd_ca_sign ]
+
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid:always,issuer
+basicConstraints = critical, CA:true, pathlen:1
+keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_CBSD
 
-ROLE=ASN1:SEQUENCE:pal_seq
+[ cps ]
 
-[ pal_seq ]
-field1 = UTF8:PAL
-field2 = UTF8:CA
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
 
 ####################################################################
-[ wif_cbsd_ca ]
+[ cbsd_oem_ca ]
 
 subjectKeyIdentifier = hash
 basicConstraints = critical, CA:true, pathlen:0
 keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_CBSD
 
-ROLE=ASN1:SEQUENCE:dm_seq
+[ cps ]
 
-[ dm_seq ]
-field1 = UTF8:CBSD
-field2 = UTF8:CA
+policyIdentifier = 2.16.840.1.114412.2.1
+ CPS.1="https://www.digicert.com/CPS"
 
-[ wif_cbsd_ca_sign ]
+[ cbsd_oem_ca_sign ]
 
 subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid:always,issuer
 basicConstraints = critical, CA:true, pathlen:0
 keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+certificatePolicies=@cps,ROLE_CA,ROLE_CBSD
```

Comment:
Yes, exactly. Since the ROLE values really represent an external policy on the certs, I think including those in the CP extension fits best. Each of the intermediate CAs has the CA Role and a profile-specific Role. Each of the end-entities has just the profile-specific Role.